spec/requests/admin/pages_spec.rb

```diff
@@ -38,6 +38,40 @@
 
       expect(page).to have_text("A$6.49 for Tim Tam")
     end
+
+    it "displays discount code when offer_code has a code" do
```

Comment:
Could you move these to `spec/requests/admin/purchases_spec.rb`? Otherwise this looks good!